kernel/time/src/lib.rs

```diff
@@ -0,0 +1,316 @@
+//! This crate contains abstractions to interact with hardware timers.
+
+#![no_std]
+
+use core::sync::atomic::{AtomicU64, AtomicUsize, Ordering};
+
+pub use core::time::Duration;
+
+static EARLY_SLEEP_FUNCTION: AtomicUsize = AtomicUsize::new(0);
+static EARLY_SLEEPER_FREQUENCY: AtomicU64 = AtomicU64::new(0);
+
+static MONOTONIC_NOW_FUNCTION: AtomicUsize = AtomicUsize::new(0);
+static INSTANT_TO_DURATION_FUNCTION: AtomicUsize = AtomicUsize::new(0);
+static DURATION_TO_INSTANT_FUNCTION: AtomicUsize = AtomicUsize::new(0);
+static MONOTONIC_FREQUENCY: AtomicU64 = AtomicU64::new(0);
+
+static REALTIME_NOW_FUNCTION: AtomicUsize = AtomicUsize::new(0);
+static REALTIME_FREQUENCY: AtomicU64 = AtomicU64::new(0);
```

Comment:
We ought to use a bit more type safety here by using the `Atomic` wrapper type from crossbeam (I use it elsewhere) to store actual `fn` references. That also prevents the undefined behavior (likely a page fault) if we try to invoke code at address 0.

Reply:
The initial values can be dummy functions that just return 0 or some other canary value.